crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -149,39 +163,37 @@ void AES_verify_service_indicator(const EVP_CIPHER_CTX *ctx, const unsigned key_
 }
 
 void AEAD_GCM_verify_service_indicator(const EVP_AEAD_CTX *ctx) {
+
   // We only have support for 128 bit and 256 bit keys for AES-GCM. AES-GCM is
   // approved only with an internal IV, see SP 800-38D Sec 8.2.2.
-  // Not the best way to write this, but the delocate parser for ARM/clang can't
-  // recognize || if statements, or switch statements for this.
-  // TODO: Update the delocate parser to be able to recognize a more readable
-  // version of this.
-   if(EVP_AEAD_key_length(ctx->aead) == 16) {
-    FIPS_service_indicator_update_state();
-   } else if(EVP_AEAD_key_length(ctx->aead) == 32) {
+  // Note: |EVP_AEAD_key_length| returns the length in bytes.
```

Comment:
I'm afraid to make those changes now before we work on the static build.

Reply:
don't worry, I'll start working on the static build right after this so I'll change it back if need be :) The thing is, I really don't see how this has anything to do with delocator, so I'm hoping it was some other issue causing delocation to fail.